components/rpill/browser/rpill.cc

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include <locale>
+#include <string>
+
+#include "brave/components/rpill/browser/rpill.h"
+
+#include "base/strings/string_util.h"
+#include "base/win/wmi.h"
+#include "content/public/browser/browser_thread.h"
+
+namespace rpill {
+
+bool exec() {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::IO);
+  base::win::WmiComputerSystemInfo info =
+      base::win::WmiComputerSystemInfo::Get();
+  auto m = info.manufacturer();
+  auto mo = info.model();
+  const std::string man(m.begin(), m.end());
```

Comment:
this is not correct for converting between wstring/string. Why are you doing this in the first place? If you want to do string comparisons on wstring use `L"string"` instead of `"string"`

Reply:
I'm also reasonably sure the string comparisons below won't work. `std::string` is encoding agnostic, but you're copying 2-byte strings and then comparing them to 1-byte strings. So  it's kind of like "virtual" != "v i r t u a l ". When you compare the actual bytes of the string they won't match.

Reply:
We're using `based::ToLowerASCII` over the result of the wmi query. So I'm converting to string first.

The conversion from `std::wstring` to `std::string` seems to be working fine:

```
    const std::string str = "testing";
    const std::string str2 = "testing2";
    const wstring wstr = L"testing" ;
    
    const std::string conv_wstr(wstr.begin(), wstr.end());

    std::cout << conv_wstr;  // "testing"
    std::cout << str.compare(conv_wstr);  // 0
    std::cout << str2.compare(conv_wstr);  // 1
```

I'm happy to refactor this logic and add unit tests, if you prefer.